test/testdata/resolver/t_let_block.rb

```diff
@@ -0,0 +1,58 @@
+# typed: true
+
+expr = T.unsafe(nil)
+MyType = T.type_alias {Integer}
+
+# For these using **expr, I don't really care what happens as long as it doesn't crash.
+
+T.let(**expr)
+T.let(**expr) {MyType}
+T.let(expr, **expr)
+T.let(expr, **expr) {MyType}
+
+T.let(checked: false, **expr)
+T.let(checked: false, **expr) {MyType}
+T.let(expr, checked: false, **expr) {MyType} # error: specific keys are unknown
+
+# --- Neither block form nor positional form ---
+
+T.let(expr) # error: No type provided to `T.let`
+T.let(expr, checked: false) # error: No type provided to `T.let`
+
+# --- Block form ---
+
+T.let(expr, checked: false, garbage: true) {MyType} # error: Unrecognized keyword argument `garbage`
+
+T.let(expr) {MyType} # error: Using `T.let` in block form requires `checked: false`
+
+T.let(expr, garbage: true) {MyType} # error: Unrecognized keyword argument `garbage`
+T.let(expr, checked: nil) {MyType} # error: Expected `T::Boolean` but found `NilClass`
+T.let(expr, checked: 1) {MyType} # error: Expected `T::Boolean` but found `Integer(1)`
+
+T.let(expr, checked: true) {MyType} # error: Using `T.let` in block form requires `checked: false`
```

Comment:
Since we require `checked: false` if we use blocks, should there be a test for non-constant `checked:` + block?  Actually, should we require a constant `checked:` argument always, block or no block?

Reply:
What do you mean “constant checked argument”

Reply:
Oh nvm re-reading I see what you mean.

we didn’t do that before, and I didn’t want to make a breaking change if we didn’t need to. Good call about writing tests for non-constant checked with a block

Reply:
I have added a test for non-literal `checked: ...` and required that the block for must have `checked: false` syntactically  (not `checked: expr` where `expr = false`)